File: backend/src/main/java/com/campussync/campussync_backend/entity/Department.java
package com.campussync.campussync_backend.entity;

import java.util.List;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotBlank;

import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "departments")
@Getter
@Setter
public class Department {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToMany(mappedBy = "department")
    private List<Student> students;

    @NotBlank
    private String name;

    @NotBlank
    private String code;

    private Boolean active;
}

File: frontend/lib/models/department.dart
class Department {
  final int id;
  final String name;
  final String code;
  final bool active;

  Department({
    required this.id,
    required this.name,
    required this.code,
    required this.active,
  });

  factory Department.fromJson(Map<String, dynamic> json) {
    return Department(
      id: json['id'],
      name: json['name'] ?? '',
      code: json['code'] ?? '',
      active: json['active'] ?? false,
    );
  }
}